### ISSUE TITLE
DNS name validator should not throw an error if an IP address is specified

### ISSUE THREAD

**Issue by jon-oracle** [digicert/pkilint#136]
I'm getting a few errors in this form:

```
GeneralNameDnsNameSyntaxValidator @ certificate.tbsCertificate.extensions.2.extnValue.subjectAltName.0.dNSName
    pkix.invalid_domain_name_syntax (ERROR): Invalid domain name syntax: "192.168.1.102"
```

The corresponding RFC does state that DNS can specify an IP address:

From https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.6

```
When the subjectAltName extension contains a domain name system
   label, the domain name MUST be stored in the dNSName (an IA5String).
   The name MUST be in the "preferred name syntax", as specified by
   [Section 3.5 of [RFC1034]](https://www.rfc-editor.org/rfc/rfc1034#section-3.5) and as modified by Section 2.1 of
   [RFC1123].
```

Then from the linked https://www.rfc-editor.org/rfc/rfc1123#page-13

```
Whenever a user inputs the identity of an Internet host, it SHOULD
      be possible to enter either (1) a host domain name or (2) an IP
      address in dotted-decimal ("#.#.#.#") form. 
```

I think this particular error should be either removed or downgraded to a WARNING at least, or maybe just an INFO?

**Comment by CBonnell** [comment]
Hi @jon-oracle, thanks for filing this.

The passage you quoted in RFC 1123 makes it clear that an quad-octet dotted IP address is discrete from a host domain name (otherwise there would be no need to enumerate these two different types). Additionally, the last paragraph of RFC 1123, section 2.1 says:

> However, a valid host name can never have the dotted-decimal form #.#.#.#, since at least the highest-level component label will be alphabetic.

Thus, it is clear that a `dNSName` cannot contain a dotted IP address, as it must contain a valid host name. This is the interpretation taken by the WebPKI, where cases of CAs encoding IPv4 addresses in `dNSName`s have been treated as non-compliance with RFC 5280.

Given this, I don't have any plans of changing the current behavior. However, I'll keep this issue open for a few more days if you have any follow-up feedback.

**Comment by CBonnell** [comment]
Closing for now, please comment if there's any follow-up.
